include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -2826,17 +2826,22 @@ class binary_reader
         {
             return;
         }
-        if constexpr(std::is_integral_v<NumberType>)
+        else if constexpr(std::is_integral_v<NumberType>)
         {
             number = std::byteswap(number);
             return;
         }
-#endif
-        auto* ptr = reinterpret_cast<std::uint8_t*>(&number);
-        for (std::size_t i = 0; i < sz / 2; ++i)
+        else
         {
-            std::swap(ptr[i], ptr[sz - i - 1]);
+#endif
+            auto* ptr = reinterpret_cast<std::uint8_t*>(&number);
+            for (std::size_t i = 0; i < sz / 2; ++i)
+            {
+                std::swap(ptr[i], ptr[sz - i - 1]);
+            }
+#ifdef __cpp_lib_byteswap
```

Comment:
Not sure if this was a style preference, or if it confused one of the formatters, but you can simplify this a little by not having the braces be inside the ifdef.  As it is here, you have the potential for the two ifdef statements to get out of sync.

Reply:
cpplint complained, and I thought "fixing" it was less effort than silencing it.